src/Features/CSharp/Portable/EditAndContinue/StatementSyntaxComparer.cs

```diff
@@ -164,6 +164,7 @@ internal enum Label
             ForStatement,
             ForStatementPart,                 // tied to parent
             ForEachStatement,
+            ForEachComponentStatement,
```

Comment:
I'd expect there should at some point. For now, I'm trying to block scenarios. I'll take a follow-up work item to confirm if distance computation is required for the scenarios I'm leaving open.